demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/query/domain/model/projection/AccountBalance.kt

```diff
@@ -1,9 +1,15 @@
 package io.holixon.cqrshexagonaldemo.demoparent.transactions.query.domain.model.projection
 
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.account.Amount
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.account.Iban
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.account.Money
 
 data class AccountBalance(
     var iban: Iban,
     var balance: Money
-)
+) {
+    fun addAmount(amount: Amount) : AccountBalance {
+        val newBalance = this.balance.amount + amount
+        return this.copy(iban = this.iban, balance = Money(newBalance))
```

Comment:
When using `copy` one only needs to provide the values to be changed. The iban remains the same, therefore it can be ommitted:
```kotlin
return this.copy(iban = this.iban, balance = Money(newBalance))
```